{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fa6ecd2c-1eae-422c-8845-138b11f94fe5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import nbformat\n",
    "from nbconvert import PythonExporter\n",
    "from nbconvert.preprocessors import ExecutePreprocessor\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fded07d1-dc2b-407b-83a6-e7cbe8e4f230",
   "metadata": {},
   "outputs": [],
   "source": [
    "mice = [\"466029_f3\", \"475165_f1\", \"475165_f3\", \"475165_f5\", \"475165_f7\"]\n",
    "days = [f\"day{d}\" for d in range(1, 16)]\n",
    "\n",
    "path_list = []\n",
    "for mouse in mice:\n",
    "    for day in days:\n",
    "        path = f\"C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_{mouse}_stage2_{day}.ipynb\"\n",
    "        path_list.append(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "332e49d4-b146-4034-bb1c-41be8cb17eb6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day1.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day1.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 47 is out of bounds for axis 0 with size 47\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day2.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day2.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day3.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day3.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day4.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day4.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "full_roi_nogo, full_nogo_segments = avg_trials(nogo_trials, smoothed_signal, double=False, segment_size=200, minus=20)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "Segment single length mismatch: expected 200, got 0 at index -2568209.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2567589.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2565986.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2563704.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2555931.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2554672.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2553741.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2551786.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2551520.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2550558.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2548669.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2546011.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2545079.0\n",
      "Segment single length mismatch: expected 200, got 0 at index -2544815.0\n",
      "No valid segments found\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[32], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m full_roi_nogo, full_nogo_segments \u001b[38;5;241m=\u001b[39m avg_trials(nogo_trials, smoothed_signal, double\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, segment_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m200\u001b[39m, minus\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m20\u001b[39m)\n",
      "\n",
      "\u001b[1;31mTypeError\u001b[0m: cannot unpack non-iterable NoneType object\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day5.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day5.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day6.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day6.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 20 is out of bounds for axis 0 with size 20\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day7.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day7.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 32 is out of bounds for axis 0 with size 32\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day8.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day8.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 45 is out of bounds for axis 0 with size 45\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day9.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day9.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day10.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day10.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 30 is out of bounds for axis 0 with size 30\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day11.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day11.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day12.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day12.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day13.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day13.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day14.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day14.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "no_artifact_timecourse = interpolate(darkf_removed_frames, artifact_indices, 5)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "interpolating artifact frames...\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[19], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m no_artifact_timecourse \u001b[38;5;241m=\u001b[39m \u001b[43minterpolate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdarkf_removed_frames\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43martifact_indices\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m5\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:222\u001b[0m, in \u001b[0;36minterpolate\u001b[1;34m(no_dark_frames, artifact_indices, stim_length)\u001b[0m\n",
      "\u001b[0;32m    219\u001b[0m     interp_indices\u001b[38;5;241m.\u001b[39mappend(group)\n",
      "\u001b[0;32m    221\u001b[0m x \u001b[38;5;241m=\u001b[39m [\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m]                                                                  \u001b[38;5;66;03m#between the two interpolated side, split into equaL segments\u001b[39;00m\n",
      "\u001b[1;32m--> 222\u001b[0m xnew \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mlinspace(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m, \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshape\u001b[49m\u001b[43m(\u001b[49m\u001b[43minterp_indices\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m)                       \u001b[38;5;66;03m#linear interpolate between the two x values, at equal increments\u001b[39;00m\n",
      "\u001b[0;32m    223\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m g \u001b[38;5;129;01min\u001b[39;00m tqdm(\u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(interp_indices[:]))):                                                 \u001b[38;5;66;03m#tqdm is loading package (GUI), loops 3 dimensions of array\u001b[39;00m\n",
      "\u001b[0;32m    224\u001b[0m     y_2d \u001b[38;5;241m=\u001b[39m [no_dark_frames[interp_indices[g][\u001b[38;5;241m0\u001b[39m]], no_dark_frames[interp_indices[g][\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]]]      \u001b[38;5;66;03m#the frames which are usable on either side of artifact\u001b[39;00m\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: tuple index out of range\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day15.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_466029_f3_stage2_day15.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "frame_number = 6000 #please insert frame number to view\n",
      "frames = load_frames(video); #loading in tif stack, this will take a little while\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "loading video, please wait...\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)\n",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_2256\\918601700.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n",
      "\u001b[0;32m      1\u001b[0m \u001b[0mframe_number\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m6000\u001b[0m \u001b[1;31m#please insert frame number to view\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m----> 2\u001b[1;33m \u001b[0mframes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mload_frames\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvideo\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m;\u001b[0m \u001b[1;31m#loading in tif stack, this will take a little while\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(video)\u001b[0m\n",
      "\u001b[0;32m     42\u001b[0m     \u001b[1;33m:\u001b[0m\u001b[1;32mreturn\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;36m3\u001b[0m\u001b[0md\u001b[0m \u001b[0marray\u001b[0m \u001b[0mof\u001b[0m \u001b[1;36m128\u001b[0m\u001b[0mx128\u001b[0m \u001b[0mx\u001b[0m \u001b[0mtimeseries\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     43\u001b[0m     \"\"\"\n",
      "\u001b[0;32m     44\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     45\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"loading video, please wait...\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m---> 46\u001b[1;33m     \u001b[0mframes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtifffile\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mimread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvideo\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m     47\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"dimensions are\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"as a\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     48\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(files, selection, aszarr, key, series, level, squeeze, maxworkers, buffersize, mode, name, offset, size, pattern, axesorder, categories, imread, sort, container, chunkshape, dtype, axestiled, ioworkers, chunkmode, fillvalue, zattrs, multiscales, omexml, out, out_inplace, _multifile, _useframes, **kwargs)\u001b[0m\n",
      "\u001b[0;32m   1074\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1075\u001b[0m         if isinstance(files, str) or not isinstance(\n",
      "\u001b[0;32m   1076\u001b[0m             \u001b[0mfiles\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcollections\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mabc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSequence\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1077\u001b[0m         \u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m-> 1078\u001b[1;33m             with TiffFile(\n",
      "\u001b[0m\u001b[0;32m   1079\u001b[0m                 \u001b[0mfiles\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1080\u001b[0m                 \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmode\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1081\u001b[0m                 \u001b[0mname\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, file, mode, name, offset, size, omexml, _multifile, _useframes, _parent, **is_flags)\u001b[0m\n",
      "\u001b[0;32m   4070\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'invalid OME-XML'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4071\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_omexml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0momexml\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4072\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_ome\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4073\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m-> 4074\u001b[1;33m         \u001b[0mfh\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mFileHandle\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moffset\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0moffset\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msize\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0msize\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m   4075\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fh\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfh\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4076\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_multifile\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0m_multifile\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0mbool\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_multifile\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4077\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_files\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[0mfh\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, file, mode, name, offset, size)\u001b[0m\n",
      "\u001b[0;32m  14210\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_offset\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0moffset\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0moffset\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14211\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_size\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0msize\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0msize\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14212\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_close\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14213\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_lock\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNullContext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m> 14214\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m  14215\u001b[0m         \u001b[1;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fh\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self)\u001b[0m\n",
      "\u001b[0;32m  14229\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32min\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;34m'rb'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'r+b'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'wb'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'xb'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14230\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33mf'\u001b[0m\u001b[1;33minvalid mode \u001b[0m\u001b[1;33m{\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14231\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrealpath\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14232\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_dir\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_name\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m> 14233\u001b[1;33m             self._fh = open(\n",
      "\u001b[0m\u001b[0;32m  14234\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14235\u001b[0m             \u001b[1;33m)\u001b[0m  \u001b[1;31m# type: ignore\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14236\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_close\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'X:\\\\RaymondLab\\\\1_Kai\\\\1_cortical_data\\\\3_third_cohort\\\\stage2\\\\466029_f3_stage2_day15.tif'\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day1.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day1.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day2.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day2.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day3.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day3.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day4.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day4.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 34 is out of bounds for axis 0 with size 34\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day5.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day5.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day6.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day6.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day7.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day7.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day8.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day8.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "frame_number = 6000 #please insert frame number to view\n",
      "frames = load_frames(video); #loading in tif stack, this will take a little while\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "loading video, please wait...\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)\n",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_2256\\918601700.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n",
      "\u001b[0;32m      1\u001b[0m \u001b[0mframe_number\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m6000\u001b[0m \u001b[1;31m#please insert frame number to view\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m----> 2\u001b[1;33m \u001b[0mframes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mload_frames\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvideo\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m;\u001b[0m \u001b[1;31m#loading in tif stack, this will take a little while\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\n",
      "\u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(video)\u001b[0m\n",
      "\u001b[0;32m     42\u001b[0m     \u001b[1;33m:\u001b[0m\u001b[1;32mreturn\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;36m3\u001b[0m\u001b[0md\u001b[0m \u001b[0marray\u001b[0m \u001b[0mof\u001b[0m \u001b[1;36m128\u001b[0m\u001b[0mx128\u001b[0m \u001b[0mx\u001b[0m \u001b[0mtimeseries\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     43\u001b[0m     \"\"\"\n",
      "\u001b[0;32m     44\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     45\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"loading video, please wait...\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m---> 46\u001b[1;33m     \u001b[0mframes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtifffile\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mimread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvideo\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m     47\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"dimensions are\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"as a\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m     48\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mframes\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(files, selection, aszarr, key, series, level, squeeze, maxworkers, buffersize, mode, name, offset, size, pattern, axesorder, categories, imread, sort, container, chunkshape, dtype, axestiled, ioworkers, chunkmode, fillvalue, zattrs, multiscales, omexml, out, out_inplace, _multifile, _useframes, **kwargs)\u001b[0m\n",
      "\u001b[0;32m   1074\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1075\u001b[0m         if isinstance(files, str) or not isinstance(\n",
      "\u001b[0;32m   1076\u001b[0m             \u001b[0mfiles\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcollections\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mabc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSequence\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1077\u001b[0m         \u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m-> 1078\u001b[1;33m             with TiffFile(\n",
      "\u001b[0m\u001b[0;32m   1079\u001b[0m                 \u001b[0mfiles\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1080\u001b[0m                 \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmode\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   1081\u001b[0m                 \u001b[0mname\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, file, mode, name, offset, size, omexml, _multifile, _useframes, _parent, **is_flags)\u001b[0m\n",
      "\u001b[0;32m   4070\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'invalid OME-XML'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4071\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_omexml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0momexml\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4072\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_ome\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4073\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m-> 4074\u001b[1;33m         \u001b[0mfh\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mFileHandle\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moffset\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0moffset\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msize\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0msize\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m   4075\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fh\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfh\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4076\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_multifile\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0m_multifile\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0mbool\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_multifile\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m   4077\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_files\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[0mfh\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, file, mode, name, offset, size)\u001b[0m\n",
      "\u001b[0;32m  14210\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_offset\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0moffset\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0moffset\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14211\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_size\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0msize\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0msize\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14212\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_close\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14213\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_lock\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNullContext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m> 14214\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m  14215\u001b[0m         \u001b[1;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fh\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;32m~\\.conda\\envs\\corticalanalysis\\Lib\\site-packages\\tifffile\\tifffile.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self)\u001b[0m\n",
      "\u001b[0;32m  14229\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32min\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;34m'rb'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'r+b'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'wb'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'xb'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14230\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33mf'\u001b[0m\u001b[1;33minvalid mode \u001b[0m\u001b[1;33m{\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14231\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrealpath\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14232\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_dir\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_name\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m> 14233\u001b[1;33m             self._fh = open(\n",
      "\u001b[0m\u001b[0;32m  14234\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_file\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14235\u001b[0m             \u001b[1;33m)\u001b[0m  \u001b[1;31m# type: ignore\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m  14236\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_close\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'X:\\\\RaymondLab\\\\1_Kai\\\\1_cortical_data\\\\3_third_cohort\\\\stage2\\\\475165_f1_stage2_day.tif'\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day9.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day9.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day10.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day10.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 33 is out of bounds for axis 0 with size 33\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day11.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day11.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day12.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day12.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day13.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day13.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day14.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day14.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day15.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f1_stage2_day15.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "full_roi_double, full_double_segments = avg_trials(double_indices, smoothed_signal, double=True, segment_size=200, minus=20)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "No valid segments found\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[28], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m full_roi_double, full_double_segments \u001b[38;5;241m=\u001b[39m avg_trials(double_indices, smoothed_signal, double\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, segment_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m200\u001b[39m, minus\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m20\u001b[39m)\n",
      "\n",
      "\u001b[1;31mTypeError\u001b[0m: cannot unpack non-iterable NoneType object\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day1.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day1.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 31 is out of bounds for axis 0 with size 31\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day2.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day2.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "double_success, double_fail = sort_success(double_txt, full_double_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[40], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m double_success, double_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdouble_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_double_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 46 is out of bounds for axis 0 with size 46\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day3.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day3.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day4.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day4.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 42 is out of bounds for axis 0 with size 42\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day5.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day5.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day6.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day6.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day7.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day7.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day8.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day8.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day9.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day9.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day10.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day10.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "txt_path = f\"X:/RaymondLab/1_Kai/2_Pi_data/{cohort}/{cageID}/stage2_{cageID}_{mouseID}/lick_rates/{session}_data.txt\"\n",
      "txt_file = load_txt(txt_path, fps=30, footer = 4)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[30], line 2\u001b[0m\n",
      "\u001b[0;32m      1\u001b[0m txt_path \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mX:/RaymondLab/1_Kai/2_Pi_data/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcohort\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcageID\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/stage2_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcageID\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mmouseID\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/lick_rates/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00msession\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_data.txt\u001b[39m\u001b[38;5;124m\"\u001b[39m\n",
      "\u001b[1;32m----> 2\u001b[0m txt_file \u001b[38;5;241m=\u001b[39m \u001b[43mload_txt\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtxt_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfps\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m30\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfooter\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m4\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:354\u001b[0m, in \u001b[0;36mload_txt\u001b[1;34m(file_path, fps, footer)\u001b[0m\n",
      "\u001b[0;32m    353\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mload_txt\u001b[39m(file_path, fps\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m30\u001b[39m, footer \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m4\u001b[39m):\n",
      "\u001b[1;32m--> 354\u001b[0m     total_lines \u001b[38;5;241m=\u001b[39m \u001b[38;5;28msum\u001b[39m(\u001b[38;5;241m1\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mfile_path\u001b[49m\u001b[43m)\u001b[49m) \u001b[38;5;66;03m# Determine the total number of lines in the file\u001b[39;00m\n",
      "\u001b[0;32m    356\u001b[0m     data_txt \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mgenfromtxt(file_path, delimiter\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m, skip_footer\u001b[38;5;241m=\u001b[39mfooter)  \u001b[38;5;66;03m# Read the text file into a NumPy array, excluding the last 3 rows\u001b[39;00m\n",
      "\u001b[0;32m    357\u001b[0m     data_footer \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mgenfromtxt(file_path, delimiter\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m, skip_header\u001b[38;5;241m=\u001b[39m(total_lines\u001b[38;5;241m-\u001b[39mfooter), filling_values\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39mnan)  \u001b[38;5;66;03m#load dark frames to line up data\u001b[39;00m\n",
      "\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'X:/RaymondLab/1_Kai/2_Pi_data/3_third_cohort/475165/stage2_475165_f3/lick_rates/475165_f3_stage2_day10_data.txt'\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day11.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day11.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 13 is out of bounds for axis 0 with size 13\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day12.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day12.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day13.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day13.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day14.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day14.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day15.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f3_stage2_day15.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "no_artifact_timecourse = interpolate(darkf_removed_frames, artifact_indices, 5)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "interpolating artifact frames...\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[19], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m no_artifact_timecourse \u001b[38;5;241m=\u001b[39m \u001b[43minterpolate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdarkf_removed_frames\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43martifact_indices\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m5\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:222\u001b[0m, in \u001b[0;36minterpolate\u001b[1;34m(no_dark_frames, artifact_indices, stim_length)\u001b[0m\n",
      "\u001b[0;32m    219\u001b[0m     interp_indices\u001b[38;5;241m.\u001b[39mappend(group)\n",
      "\u001b[0;32m    221\u001b[0m x \u001b[38;5;241m=\u001b[39m [\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m]                                                                  \u001b[38;5;66;03m#between the two interpolated side, split into equaL segments\u001b[39;00m\n",
      "\u001b[1;32m--> 222\u001b[0m xnew \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mlinspace(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m, \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshape\u001b[49m\u001b[43m(\u001b[49m\u001b[43minterp_indices\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m)                       \u001b[38;5;66;03m#linear interpolate between the two x values, at equal increments\u001b[39;00m\n",
      "\u001b[0;32m    223\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m g \u001b[38;5;129;01min\u001b[39;00m tqdm(\u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(interp_indices[:]))):                                                 \u001b[38;5;66;03m#tqdm is loading package (GUI), loops 3 dimensions of array\u001b[39;00m\n",
      "\u001b[0;32m    224\u001b[0m     y_2d \u001b[38;5;241m=\u001b[39m [no_dark_frames[interp_indices[g][\u001b[38;5;241m0\u001b[39m]], no_dark_frames[interp_indices[g][\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]]]      \u001b[38;5;66;03m#the frames which are usable on either side of artifact\u001b[39;00m\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: tuple index out of range\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day1.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day1.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day2.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day2.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day3.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day3.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day4.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day4.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "darkf_removed = remove_dark_frames(frames, mean_timecourse, 0.35)\n",
      "darkf_removed_timecourse, darkf_removed_frames = darkf_removed[0], darkf_removed[1] #first one is the timecourse with no 3d, second is the 3d array\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[16], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m darkf_removed \u001b[38;5;241m=\u001b[39m \u001b[43mremove_dark_frames\u001b[49m\u001b[43m(\u001b[49m\u001b[43mframes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmean_timecourse\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m0.35\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;32m      2\u001b[0m darkf_removed_timecourse, darkf_removed_frames \u001b[38;5;241m=\u001b[39m darkf_removed[\u001b[38;5;241m0\u001b[39m], darkf_removed[\u001b[38;5;241m1\u001b[39m] \u001b[38;5;66;03m#first one is the timecourse with no 3d, second is the 3d array\u001b[39;00m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:134\u001b[0m, in \u001b[0;36mremove_dark_frames\u001b[1;34m(frames, signal, threshold)\u001b[0m\n",
      "\u001b[0;32m    131\u001b[0m dark_frame_threshold \u001b[38;5;241m=\u001b[39m min_value \u001b[38;5;241m+\u001b[39m threshold \u001b[38;5;241m*\u001b[39m min_value     \u001b[38;5;66;03m# Calculate the threshold as the minimum value plus 10%\u001b[39;00m\n",
      "\u001b[0;32m    133\u001b[0m brain_indices \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mwhere(signal \u001b[38;5;241m>\u001b[39m dark_frame_threshold)[\u001b[38;5;241m0\u001b[39m]\n",
      "\u001b[1;32m--> 134\u001b[0m start_index \u001b[38;5;241m=\u001b[39m \u001b[43mbrain_indices\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m1\u001b[39m\n",
      "\u001b[0;32m    135\u001b[0m end_index \u001b[38;5;241m=\u001b[39m brain_indices[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\n",
      "\u001b[0;32m    137\u001b[0m frames \u001b[38;5;241m=\u001b[39m frames[start_index:end_index, :, :]\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 0 is out of bounds for axis 0 with size 0\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day5.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day5.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day6.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day6.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day7.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day7.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day8.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day8.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day9.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day9.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 11 is out of bounds for axis 0 with size 11\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day10.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day10.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day11.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day11.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 4 is out of bounds for axis 0 with size 4\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day12.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day12.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day13.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day13.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day14.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day14.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day15.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f5_stage2_day15.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "full_roi_double, full_double_segments = avg_trials(double_indices, smoothed_signal, double=True, segment_size=200, minus=20)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "No valid segments found\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[28], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m full_roi_double, full_double_segments \u001b[38;5;241m=\u001b[39m avg_trials(double_indices, smoothed_signal, double\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, segment_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m200\u001b[39m, minus\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m20\u001b[39m)\n",
      "\n",
      "\u001b[1;31mTypeError\u001b[0m: cannot unpack non-iterable NoneType object\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day1.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day1.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day2.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day2.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day3.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day3.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "nogo_success, nogo_fail = sort_success(nogo_txt, full_nogo_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[41], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m nogo_success, nogo_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnogo_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_nogo_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 12 is out of bounds for axis 0 with size 12\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day4.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day4.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "double_success, double_fail = sort_success(double_txt, full_double_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[40], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m double_success, double_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdouble_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_double_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 35 is out of bounds for axis 0 with size 35\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day5.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day5.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day6.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day6.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day7.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day7.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 29 is out of bounds for axis 0 with size 29\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day8.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day8.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day9.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day9.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day10.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day10.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 1 is out of bounds for axis 0 with size 1\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day11.ipynb\n",
      "Executed successfully: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day11.ipynb\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day12.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day12.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "single_success, single_fail = sort_success(single_txt, full_single_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[39], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m single_success, single_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43msingle_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_single_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:594\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    592\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(single_or_double_success_txt)):\n",
      "\u001b[0;32m    593\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[1;32m--> 594\u001b[0m         success\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    595\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[0;32m    596\u001b[0m         fail\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 42 is out of bounds for axis 0 with size 42\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day13.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day13.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "no_artifact_timecourse = interpolate(darkf_removed_frames, artifact_indices, 5)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "interpolating artifact frames...\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[19], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m no_artifact_timecourse \u001b[38;5;241m=\u001b[39m \u001b[43minterpolate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdarkf_removed_frames\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43martifact_indices\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m5\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:222\u001b[0m, in \u001b[0;36minterpolate\u001b[1;34m(no_dark_frames, artifact_indices, stim_length)\u001b[0m\n",
      "\u001b[0;32m    219\u001b[0m     interp_indices\u001b[38;5;241m.\u001b[39mappend(group)\n",
      "\u001b[0;32m    221\u001b[0m x \u001b[38;5;241m=\u001b[39m [\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m]                                                                  \u001b[38;5;66;03m#between the two interpolated side, split into equaL segments\u001b[39;00m\n",
      "\u001b[1;32m--> 222\u001b[0m xnew \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mlinspace(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m6\u001b[39m, \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshape\u001b[49m\u001b[43m(\u001b[49m\u001b[43minterp_indices\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m)                       \u001b[38;5;66;03m#linear interpolate between the two x values, at equal increments\u001b[39;00m\n",
      "\u001b[0;32m    223\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m g \u001b[38;5;129;01min\u001b[39;00m tqdm(\u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mlen\u001b[39m(interp_indices[:]))):                                                 \u001b[38;5;66;03m#tqdm is loading package (GUI), loops 3 dimensions of array\u001b[39;00m\n",
      "\u001b[0;32m    224\u001b[0m     y_2d \u001b[38;5;241m=\u001b[39m [no_dark_frames[interp_indices[g][\u001b[38;5;241m0\u001b[39m]], no_dark_frames[interp_indices[g][\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]]]      \u001b[38;5;66;03m#the frames which are usable on either side of artifact\u001b[39;00m\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: tuple index out of range\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day14.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day14.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "nogo_success, nogo_fail = sort_success(nogo_txt, full_nogo_segments)\n",
      "------------------\n",
      "\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[41], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m nogo_success, nogo_fail \u001b[38;5;241m=\u001b[39m \u001b[43msort_success\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnogo_txt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_nogo_segments\u001b[49m\u001b[43m)\u001b[49m\n",
      "\n",
      "File \u001b[1;32m~\\Documents\\GitHub\\corticalanalysis\\Notebook\\2_stage2\\third_cohort\\auto\\preprocessing_functions.py:596\u001b[0m, in \u001b[0;36msort_success\u001b[1;34m(single_or_double_success_txt, full_segments)\u001b[0m\n",
      "\u001b[0;32m    594\u001b[0m     success\u001b[38;5;241m.\u001b[39mappend(full_segments[i])\n",
      "\u001b[0;32m    595\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m single_or_double_success_txt[i] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "\u001b[1;32m--> 596\u001b[0m     fail\u001b[38;5;241m.\u001b[39mappend(\u001b[43mfull_segments\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m)\n",
      "\u001b[0;32m    597\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;32m    598\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtxt success rate contains number that is not 1 or 0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\n",
      "\u001b[1;31mIndexError\u001b[0m: index 26 is out of bounds for axis 0 with size 26\n",
      "\n",
      "Running: C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day15.ipynb\n",
      "Error while executing C:/Users/trapped/Documents/GitHub/corticalanalysis/Notebook/2_stage2/third_cohort/auto/auto_475165_f7_stage2_day15.ipynb: An error occurred while executing the following cell:\n",
      "------------------\n",
      "full_roi_double, full_double_segments = avg_trials(double_indices, smoothed_signal, double=True, segment_size=200, minus=20)\n",
      "------------------\n",
      "\n",
      "----- stdout -----\n",
      "No valid segments found\n",
      "------------------\n",
      "\n",
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)\n",
      "Cell \u001b[1;32mIn[28], line 1\u001b[0m\n",
      "\u001b[1;32m----> 1\u001b[0m full_roi_double, full_double_segments \u001b[38;5;241m=\u001b[39m avg_trials(double_indices, smoothed_signal, double\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, segment_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m200\u001b[39m, minus\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m20\u001b[39m)\n",
      "\n",
      "\u001b[1;31mTypeError\u001b[0m: cannot unpack non-iterable NoneType object\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Path to save executed notebooks\n",
    "output_directory = \"executed\"\n",
    "os.makedirs(output_directory, exist_ok=True)\n",
    "\n",
    "# Execution function\n",
    "def run_notebook(notebook_path, output_path):\n",
    "    try:\n",
    "        with open(notebook_path, \"r\", encoding=\"utf-8\") as f:\n",
    "            nb = nbformat.read(f, as_version=4)\n",
    "\n",
    "        # Configure notebook execution\n",
    "        ep = ExecutePreprocessor(timeout=600, kernel_name=\"python3\")\n",
    "        ep.preprocess(nb, {\"metadata\": {\"path\": os.path.dirname(notebook_path) or \"./\"}})\n",
    "\n",
    "        # Save executed notebook\n",
    "        with open(output_path, \"w\", encoding=\"utf-8\") as f:\n",
    "            nbformat.write(nb, f)\n",
    "        print(f\"Executed successfully: {notebook_path}\")\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"Error while executing {notebook_path}: {e}\")\n",
    "\n",
    "# Execute each trial notebook sequentially\n",
    "for notebook in path_list:\n",
    "    output_path = os.path.join(output_directory, f\"executed_{os.path.basename(notebook)}\")\n",
    "    print(f\"Running: {notebook}\")\n",
    "    run_notebook(notebook, output_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82c29d9d-7cd2-427a-abc3-2a7d4c272a78",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
